src/common/conditional_compilation/docs/develop_cc_for_new_component.md

```diff
@@ -0,0 +1,209 @@
+# How to enable conditional compilation for new component
+
+There are several macros to help enable conditional compilation(CC) feature for new component, and in each CC mode(SELECTIVE_BUILD_ANALYZER and SELECTIVE_BUILD) they must be defined with same name but difference macro content definition. Developer can apply or reference these macros for own component's conditional compilation enablement. You can find these macros in [selective_buld.h](https://github.com/openvinotoolkit/openvino/blob/master/src/common/conditional_compilation/include/openvino/cc/selective_build.h) and [itt.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/conditional_compilation/include/openvino/cc/pass/itt.hpp).
+
+## Introduction of conditional compilation macro
+
+### 1. Macro in SELECTIVE_BUILD_ANALYZER mode
+#### OV_CC_DOMAINS
+It defines domains for conditional compilation, in which contains 3 macros to define detail domains(simple scope, switch/cases and factories). The implement code is:
+```ruby
+    #  define OV_CC_DOMAINS(Module)                                                                   \
```

Comment:
Can we use references to real code? like here: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/ir/docs/support_new_opset.md

Reply:
Good idea!

Reply:
done